File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.java
class Solution {
    public int[] productExceptSelf(int[] nums) {
        int productOfAll = 1;
        for(int num: nums){
            productOfAll = productOfAll * num;
        }
        int[] res = new int[nums.length];
        for(int i=0;i<nums.length;i++){
            if(nums[i]==0){
                int tmp = 1;
                for(int j=0;j<nums.length;j++){
                    if(j!=i){
                        tmp = tmp*nums[j];
                    }
                }
                res[i] = tmp;
                continue;
            }
            res[i] = productOfAll / nums[i];
        }
        return res;
    }
}  

File: Data Structures & Algorithms/subsets/submission-0.py
class Solution:
    def subsets(self, nums: List[int]) -> List[List[int]]:
        res = []
        subset = []
        def dfs(i):
            # if subset not in res:
            #     res.add(subset)
            if i==len(nums):
                res.append(subset.copy())
                return
            
            # decide to include nums[i]
            subset.append(nums[i])
            dfs(i+1)
            # decide not to include nums[i]
            subset.pop()
            dfs(i+1)

            # for x in nums:
            #     if x in subset:
            #         continue
            #     tmp = (x)
            #     subset = subset + tmp
            #     dfs(subset, step+1)
            #     subset= subset-tmp
        dfs(0)
        return res

